types/underscore/underscore-tests.ts

```diff
@@ -165,372 +165,423 @@ declare const level2UnionList: _.List<_.List<string | number>>;
-var throttled = _.throttle(updatePosition, 100);
-$(window).scroll(throttled);
-throttled.cancel();
+{
+    const updatePosition = (param: string) => alert('updating position... Param: ' + param);
+    const throttled = _.throttle(updatePosition, 100); // $ExpectType ((param: string) => void) & Cancelable
+    $(window).scroll(throttled);
+    throttled.cancel(); // $ExpectType void
+}
 
 // debouncing a function
-var calculateLayout = (param:string) => alert('calculating layout... Param: ' + param);
-var lazyLayout = _.debounce(calculateLayout, 300);
-$(window).resize(lazyLayout);
-lazyLayout.cancel();
+{
+    const calculateLayout = (param: string) => alert('calculating layout... Param: ' + param);
+    const lazyLayout = _.debounce(calculateLayout, 300); // $ExpectType ((param: string) => void) & Cancelable
+    $(window).resize(lazyLayout);
+    lazyLayout.cancel(); // $ExpectType void
+}
 
 // creating a function that will only perform its action once (the second call will return the result of the first call)
-var createApplication = (param:string) => alert('creating application... Param: ' + param);
-var initialize = _.once(createApplication);
-initialize("me");
-initialize("me");
+{
+    const createApplication = (param: string) => 'creating application... Param: ' + param;
+    const initialize = _.once(createApplication); // $ExpectType (param: string) => string
+    initialize("first"); // $ExpectType string
+    initialize("second"); // $ExpectType string
+}
 
 // creating a wrapped function that will only be invoked after the wrapper is invoked a number of times
-var notes: any[] = [1,2,3];
-var render = () => alert("rendering...");
-var renderNotes = _.after(notes.length, render);
-_.each(notes, (note) => note.asyncSave({ success: renderNotes }));
+{
+    const renderNotes = _.after(anyArray.length, () => alert("rendering...")); // $ExpectType Function
+    _.each(anyArray, note => note.asyncSave({ success: renderNotes })); // $ExpectType any[]
+}
 
 // wrapping a function in another function
-var hello = function (name) { return "hello: " + name; };
-var wrappedHello = _.wrap(hello, (func) => { return `before, ${func("moe")} + after`; });
-wrappedHello();
+{
+    const hello = (name: string) => "hello: " + name;
+    const hello2 = _.wrap(hello, func => `before, ${func("moe")} + after`); // $ExpectType Function
+    hello2(); // $ExpectType any
+}
 
 // composing a function as the result of multiple function calls
-var greet = function (name) { return "hi: " + name; };
-var exclaim = function (statement) { return statement + "!"; };
-var welcome = _.compose(exclaim, greet);
-welcome('moe');
-
-var partialApplicationTestFunction = (a: string, b: number, c: boolean, d: string, e: number, f: string) => {  }
+{
+    const greet = (name: string) => "hi: " + name;
+    const exclaim = (statement: string) => statement + "!";
+    const welcome = _.compose(exclaim, greet); // $ExpectType Function
+    welcome('moe'); // $ExpectType any
```

Comment:
I agree, the `Function` family of Underscore functions could also use some typing updates since a lot of them use `Function` and `...args: any[]` rather than type checking parameters. It was nice for my own usages to take care of the Collection and Array families first, though, and it was also nice to wait to touch these until TS3.1 became the new minimum version since that provides useful additional support for a lot of these operations.